pkg/asset/installconfig/alibabacloud/validation.go

```diff
@@ -100,6 +119,56 @@ func validateResourceGroup(client *Client, ic *types.InstallConfig, path *field.
 	return append(allErrs, field.NotFound(path.Child("resourceGroupID"), ic.AlibabaCloud.ResourceGroupID))
 }
 
+func validateVpc(client *Client, ic *types.InstallConfig, path *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	vpcs, err := client.ListVpcs(ic.AlibabaCloud.VpcID)
+	if err != nil {
+		return append(allErrs, field.InternalError(path.Child("vpcID"), err))
+	}
+	if vpcs.TotalCount == 1 {
+		return allErrs
+	}
+	return append(allErrs, field.NotFound(path.Child("vpcID"), ic.AlibabaCloud.VpcID))
+}
+
+func validateVSwitchs(client *Client, ic *types.InstallConfig, path *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if ic.AlibabaCloud.VpcID == "" {
+		return append(allErrs, field.Required(path.Child("vswitchIDs"), "Use the existing VSwitchs, vpcID must be specified"))
+	}
+
+	for _, id := range ic.AlibabaCloud.VSwitchIDs {
```

Comment:
Use an index variable so that the paths used in the field errors can reference the index.
Example,
```
allErrs = append(allErrs, field.NotFound(path.Child("vswitchIDs").Index(i), id))
```